addons/test_mail/static/tests/systray_activity_menu_tests.js

```diff
@@ -1,205 +1,140 @@
-        assert.hasClass(second, "fa-clock-o", "should display the activity action icon");
-
-        patchWithCleanup(env.services.action, {
-            doAction(action) {
-                if (action.name) {
-                    assert.ok(action.domain, "should define a domain on the action");
-                    assert.deepEqual(
-                        action.domain,
-                        [["activity_ids.user_id", "=", 10]],
-                        "should set domain to user's activity only"
-                    );
-                    assert.step("do_action:" + action.name);
-                } else {
-                    assert.step("do_action:" + action);
-                }
+            {
+                date_deadline: date_to_str(yesterday),
+                res_id: mailTestActivityIds[3],
+                res_model: "mail.test.activity",
             },
-        });
-
-        assert.hasClass(
-            document.querySelector(".o-dropdown-menu"),
-            "show",
-            "dropdown should be expanded"
-        );
+        ]);
+    },
+});
 
-        await click(
-            '.o_ActivityMenuView_activityGroupActionButton[data-model_name="mail.test.activity"]'
-        );
-        assert.containsNone(document.body, ".o-dropdown-menu", "dropdown should be collapsed");
+QUnit.test("activity menu widget: menu with no records", async function (assert) {
+    assert.expect(1);
 
-        // click on the "res.partner" activity icon
-        await click(".o_menu_systray .dropdown-toggle:has(i[aria-label='Activities'])");
-        await click('.o_ActivityMenuView_activityGroupActionButton[data-model_name="res.partner"]');
+    const { click } = await start({
+        mockRPC: function (route, args) {
+            if (args.method === "systray_get_activities") {
+                return Promise.resolve([]);
+            }
+        },
+    });
+    await click(".o_menu_systray .dropdown-toggle:has(i[aria-label='Activities'])");
+    assert.containsOnce(target, ".o-mail-no-activity");
+});
 
-        assert.verifySteps(["do_action:mail.test.activity", "do_action:res.partner"]);
+QUnit.test("activity menu widget: activity menu with 2 models", async function (assert) {
+    const { env } = await start();
+    assert.containsOnce(target, ".o_menu_systray i[aria-label='Activities']");
+    assert.containsOnce(target, ".o-mail-activity-menu-counter");
+    assert.strictEqual($(target).find(".o-mail-activity-menu-counter").text(), "5");
+    var context = {};
+    patchWithCleanup(env.services.action, {
+        doAction(action) {
+            assert.deepEqual(action.context, context);
+        },
     });
+    context = {
+        force_search_count: 1,
+        search_default_activities_overdue: 1,
+    };
+    await click(".o_menu_systray i[aria-label='Activities']");
+    assert.containsOnce(target, ".o-mail-activity-menu");
+    assert.ok(target.querySelectorAll(".o-mail-activity-menu .o-mail-activity-group").length);
```

Comment:
Assertion of `ok(length)` is not specific enough. Better tell how many there are.